### PR TITLE
`ListUsers` API protobuf definitions

### PR DESCRIPTION
**Pull request by willvedd**
## Description

Defining protobufs to facilitate OpenFGA's upcoming `ListUsers` functionality which answers the question of "what users relate to a specific resource?". The definitions themselves **are not final** and are subject to change. Please refer to the [ListUsers RFC](https://github.com/openfga/rfcs/pull/15) for more specifics about the API's functionality as well as technical changes that can be expected.

**Note:** This PR is a draft to enable `ListUsers` development in the main OpenFGA repo.

## References
- [ListUsers RFC](https://github.com/openfga/rfcs/pull/15)

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: openfga/v1/openfga_service.proto [hunk 1]
```diff
@@ -731,6 +745,53 @@ message ListObjectsResponse {
...
+  ];
+
+  string relation = 5 [
+    (validate.rules).string = {pattern: "^[^:#@\\s]{1,50}$"},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"reader\""}
+  ];
+
+  repeated RelationReference filters = 6 [
+    json_name = "filters",
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:779]:
```suggestion
  repeated RelationReference user_filters = 6 [
    json_name = "user_filters",
```

File: openfga/v1/openfga_service.proto [hunk 2]
```diff
@@ -731,6 +745,53 @@ message ListObjectsResponse {
...
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""}
+  ];
+
+  Object object = 3;
+
+  repeated string target_user_object_types = 4 [
+    json_name = "targetUserObjectTypes",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "[\"user\",\"employee\"]"},
+    (google.api.field_behavior) = REQUIRED
+  ];
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:770]: you don't need this, it's the same as `filters`

File: openfga/v1/openfga_service.proto [hunk 3]
```diff
@@ -731,6 +745,53 @@ message ListObjectsResponse {
...
+  string relation = 5 [
+    (validate.rules).string = {pattern: "^[^:#@\\s]{1,50}$"},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"reader\""}
+  ];
+
+  repeated RelationReference filters = 6 [
+    json_name = "filters",
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: ""}
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:781]:
```suggestion
    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example:  "[\"user\",\"group#member\"]"}
```

File: openfga/v1/openfga_service.proto [hunk 4]
```diff
@@ -731,6 +745,53 @@ message ListObjectsResponse {
...
+  string authorization_model_id = 2 [
+    json_name = "authorization_model_id",
+    (validate.rules).string = {
+      pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$",
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""}
+  ];
+
+  Object object = 3;
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:764]:
```suggestion
  Object object = 3 [
    json_name = "object",
    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"document:budget\""},
    (google.api.field_behavior) = REQUIRED
  ];
  
```

File: openfga/v1/openfga_service.proto [hunk 5]
```diff
@@ -731,6 +745,103 @@ message ListObjectsResponse {
...
+  string relation = 4 [
+    (validate.rules).string = {pattern: "^[^:#@\\s]{1,50}$"},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"reader\""}
+  ];
+
+  repeated ListUsersFilter user_filters = 5 [
+    json_name = "user_filters",
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "[{\"type\": \"user\"}, {\"type\": \"SetUser\", \"relation\": \"member\"}]"}
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:779]:
```suggestion
    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "[{\"type\": \"user\"}, {\"type\": \"group\", \"relation\": \"member\"}]"}
```

**Review by miparnisari — APPROVED** [review]

File: openfga/v1/openfga_service.proto [hunk 6]
```diff
@@ -731,6 +745,103 @@ message ListObjectsResponse {
...
+    (validate.rules).string = {pattern: "^[^:#@\\s]{1,50}$"},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"reader\""}
+  ];
+
+  repeated ListUsersFilter user_filters = 5 [
+    json_name = "user_filters",
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "[{\"type\": \"user\"}, {\"type\": \"group\", \"relation\": \"member\"}]"}
+  ];
```
Comment (miparnisari) [inline openfga/v1/openfga_service.proto:780]: max size = 1?

Comment (willvedd) [reply openfga/v1/openfga_service.proto:780]: Probably a good idea. I think we could probably support > 1 but better to constrain and remove than to add later.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**